### ISSUE TITLE
partially decoded special chars in query params

### ISSUE THREAD

**Issue by georgecrawford** [sindresorhus/normalize-url#26]
`'https://example.com?tint=%23505050,%23505050'` is normalised to `'https://example.com/?tint=%23505050,#505050'`. Note that only one of the entities is decoded. I'm not even sure that the entities should be decoded here - what's the justification for that?

**Comment by sindresorhus** [comment]
Would you mind adding a [failing test](https://github.com/avajs/ava#failing-tests)?

**Comment by georgecrawford** [comment]
Gladly. But could you explain what you'd prefer to see? Both `%23`s transformed to `#`, or neither?

**Comment by sindresorhus** [comment]
It should decode both.

Relevant: https://github.com/sindresorhus/normalize-url/blob/4fdef83bdc2523c74f87e47329af938cff91e6b4/test.js#L16

**Comment by georgecrawford** [comment]
Uh-oh:
```
$ node
> url = {search:'foo=#505050,#505050'}
{ search: 'foo=#505050,#505050' }
> require('url').format(url)
'?foo=%23505050,#505050'
```
Do you think this is a Node bug? Seems to affect Node 4, 5 and 6.

**Comment by georgecrawford** [comment]
https://github.com/sindresorhus/normalize-url/pull/27

**Comment by sindresorhus** [comment]
Yes, looks like a Node.js bug. Can you open an issue over at the Node.js issue tracker and paste a link to it here?

**Comment by ben-eb** [comment]
Looks like it's fixed! Using `v4.7.0`:

```
$ node
> url = {search:'foo=#505050,#505050'}
{ search: 'foo=#505050,#505050' }
> require('url').format(url)
'?foo=%23505050,%23505050'
```
